packages/app/src/systems/SignUp/components/MnemonicConfirm/MnemonicConfirm.tsx

```diff
@@ -0,0 +1,247 @@
+import { cssObj } from '@fuel-ui/css';
+import { Box, Flex, Grid, Button, Stack, Alert } from '@fuel-ui/react';
+import { useEffect, useMemo, useState } from 'react';
+
+import { MnemonicInput } from '../../../Core/components/Mnemonic/MnemonicInput';
+import { Header } from '../Header';
+
+import { ImageLoader, relativeUrl } from '~/systems/Core';
+
+function fillArray(item: string[], format: number) {
+  return Array.from({ length: format }).map((_, idx) => item[idx] || '');
+}
+
+function checkMoreThanOneWord(word: string) {
+  if (word.split(' ').length > 1) {
+    const first = word.split(' ')[0];
+    const half = first.slice(0, first.length / 2);
+    return `${half}${half}` === first ? half : first;
+  }
+  return word;
+}
+
+export type MnemonicConfirmProps = {
+  words?: string[];
+  onFilled?: (val: string[]) => void;
+  onNext: () => void;
+  onCancel: () => void;
+  positions?: number[];
+  readOnly?: boolean;
+  defaultValue?: string[];
+  isLoading?: boolean;
+  canProceed?: boolean;
+  error?: string | boolean;
+};
+
+export function MnemonicConfirm({
+  words = [],
+  onFilled,
+  onNext,
+  onCancel,
+  positions,
+  readOnly,
+  defaultValue = fillArray([], 9),
```

Comment:
```suggestion
  defaultValue = fillArray([], NUM_WORDS_TO_CONFIRM),
```
this isn't necessary but we could add
```
export const NUM_WORDS_TO_CONFIRM = 9
```
to the top of this file and then import and use the const in the `MnemonicConfirm` test.  That way if we ever change this number the tests won't break